monitoring/v3/src/main/java/ListResources.java

```diff
@@ -0,0 +1,211 @@
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.monitoring.v3.Monitoring;
+import com.google.api.services.monitoring.v3.MonitoringScopes;
+import com.google.api.services.monitoring.v3.model.ListMetricDescriptorsResponse;
+import com.google.api.services.monitoring.v3.model.ListMonitoredResourceDescriptorsResponse;
+import com.google.api.services.monitoring.v3.model.ListTimeSeriesResponse;
+
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.security.GeneralSecurityException;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+
+
+
+/**
+ * Simple command-line program to demonstrate connecting to and retrieving data
+ * from the Google Cloud Monitoring API v3 using application default credentials.
+ */
+public class ListResources {
+
+  /**
+   * The metric that we want to fetch.
+   */
+  private static final String METRIC =
+      "compute.googleapis.com/instance/cpu/usage_time";
+
+  /**
+   * This test program prints to standard output, but the integration tests verify
+   * the output with a custom output stream.
+   */
+  private PrintStream outputStream;
+
+  /**
+   * Cloud Monitoring v3 REST client.
+   */
+  private Monitoring monitoringService;
+
+  /**
+   * Identifier for project resource, in format 'projects/your-project-id'.
+   */
+  private String projectResource;
+
+  /**
+   * Utility class doesn't need to be instantiated.
+   */
+  private ListResources(Monitoring monitoringService, String projectResource) {
+    this.monitoringService = monitoringService;
+    this.projectResource = projectResource;
+    this.outputStream = System.out;
+  }
+
+  /**
+   * Package private that accepts output stream for integration test.
+   */
+  ListResources(Monitoring monitoringService, String projectResource, PrintStream os) {
+    this.monitoringService = monitoringService;
+    this.projectResource = projectResource;
+    this.outputStream = os;
+  }
+
+  private static SimpleDateFormat rfc3339 =
+      new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'");
+
+  static {
+    rfc3339.setTimeZone(TimeZone.getTimeZone("UTC"));
+  }
+
+  /**
+   * Query the projects.monitoredResourceDescriptors.list API method.
+   * This lists all the resources available to be monitored in the API.
+   * <p/>
+   * Package-private to be accessible to tests.
+   */
+  void listMonitoredResourceDescriptors() throws IOException {
+    ListMonitoredResourceDescriptorsResponse monitoredResources =
+        this.monitoringService.projects()
+            .monitoredResourceDescriptors().list(this.projectResource).execute();
+    this.outputStream.println("listMonitoredResourceDescriptors response");
+    this.outputStream.println(monitoredResources.toPrettyString());
+  }
+
+  /**
+   * Query to MetricDescriptors.list
```

Comment:
`{@link}` for MetricDescriptors.list. I think formatted as `MetricDescriptors#list`, if I remember the format correctly.